### PR TITLE
Add samuel's literate keymap file for planck

### PR DESCRIPTION
**Pull request by asamwow**
## Description
Added my awesome keymap written in a .org file for emacs users mostly. This way, you can easily edit your keymap in the org mode table, and changes will be reflected in the actual tethered keymap file. No more having to maintain comments and balancing tables to have a neat config file.

The actual keymap is not that important, just allowing other's to easily create their own keymaps in org if they wish. But if you are interested in the keymap, it is heavily based on a standard dvorak layout using a single extra layer and a lot of custom implemented tap keys to make it all work. It feels amazing!!!

## Types of changes
- [ ] Core
- [ ] Bugfix
- [ ] New Feature
- [ ] Enhancement/Optimization
- [ ] Keyboard (addition or update)
- [X] Keymap/Layout/Userspace (addition or update)
- [ ] Documentation

## Checklist:
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document. (https://docs.qmk.fm/#/contributing)
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/planck/keymaps/samuel/keymap.c [hunk 1]
```diff
@@ -0,0 +1,298 @@
+#include "planck.h"
```
Comment (drashna) [inline keyboards/planck/keymaps/samuel/keymap.c]: Could you replace this:
```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/planck/keymaps/samuel/keymap.c [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+#include "action_layer.h"
```
Comment (drashna) [inline keyboards/planck/keymaps/samuel/keymap.c]: Could you remove this additional include?  

```suggestion
```

They get included automatically, at various points in the built process, so they are redundant and not necessary here.

File: keyboards/planck/keymaps/samuel/keymap.c [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+
+#define TAPPING_TERM 250
```
Comment (drashna) [inline keyboards/planck/keymaps/samuel/keymap.c]: This must be in the `config.h` file, or it won't actually get honored.

File: keyboards/planck/keymaps/samuel/keymap.c [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+break;
+case KC_T_RALT:
+if (record->event.pressed) {
+  tap_timers[1] = timer_read();
+  tap_interupted = false;
+  register_mods(MOD_BIT(KC_RALT));
+ } else {
+  unregister_mods(MOD_BIT(KC_RALT));
+  if (!tap_interupted && timer_elapsed(tap_timers[1]) < TAPPING_TERM) {
+    mod_type(KC_LSHIFT, KC_0);
```
Comment (drashna) [inline keyboards/planck/keymaps/samuel/keymap.c]: Actually, I think you can do `tap_code16(S(KC_0))` here

**Comment by asamwow** [comment]
Hello, please take as long as you need for this PR, but I just wanted to communicate that my last commit addressed issues raised in @drashna 's review. Please tell me if anything still needs to be looked at again, thank you.

**Comment by drashna** [comment]
The only remaining issue is a coding convention (formatting) one.  As is, it's pretty hard to read.  If you're fine with that, then that's okay.

And yes, we squash keymap PRs, like this.

**Comment by asamwow** [comment]
I am fine with the formatting. I and other emacs users if they so choose to use this keymap will never look at the keymap file, but only at the org file. Thanks!

**Pull request merged**